src/transcription/utils.ts

```diff
@@ -0,0 +1,206 @@
+import { AxiosInstance } from 'axios'
+import FormData from 'form-data'
+import { createReadStream, existsSync } from 'node:fs'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+import { TranscribeRequest } from './types'
+
+interface UploadFileResponse {
+  url: string
+}
+
+/**
+ * Transforms a transcribe request to the API-expected format.
+ *
+ * Maps `source` to `input.url` and converts translation arrays (if present)
+ * into comma-separated strings.
+ *
+ * @param transcribeRequest - The original transcribe request.
+ * @returns The transformed request object.
+ */
+export const transformTranscribeRequest = (transcribeRequest: TranscribeRequest): Record<string, any> => {
+  const { source, organizationName, options, ...rest } = transcribeRequest
+
+  const transformedOptions = options
+    ? {
+        ...options,
+        llm_translation: options.llmTranslation?.join(', '),
+        srt_translation: options.srtTranslation?.join(', '),
+      }
+    : {}
+
+  return {
+    ...rest,
+    input: {
+      url: source,
+      ...transformedOptions,
+    },
+  }
+}
+
+/**
+ * Normalizes a file path string to an absolute path.
+ *
+ * Handles:
+ * - File URLs (e.g. "file:///C:/path/to/file.txt")
+ * - Tilde expansion for home directories (e.g. "~/file.txt")
+ * - Environment variables (e.g. "$HOME/file.txt" or "%APPDATA%\\file.txt")
+ * - Relative paths (resolved against process.cwd())
+ *
+ * @param filePath - The file path string.
+ * @returns A normalized absolute file path.
+ */
+const normalizeFilePath = (filePath: string): string => {
```

Comment:
This should probably have unit tests with a wide range of path use cases. Aside from validating the function, it should also serve as documentation of the range of inputs that are acceptable?

Reply:
I split the project into tasks (PRs). Once the existing function-related PRs are merged, I'll add the unit tests covering a wide range of path use cases, serving both as validation and documentation.